terraform/nomad-whisper/create-nomad-whisper-instance.sh

```diff
@@ -0,0 +1,241 @@
+
+# look up instance pool. If it exists, find its curent size and set INSTANCE_POOL_SIZE appropriately
+INSTANCE_POOL_DETAILS="$(oci compute-management instance-pool list --region "$ORACLE_REGION" -c "$COMPARTMENT_OCID" --all --display-name "$INSTANCE_POOL_NAME" | jq '.data[0]')"
+
+if [ -z "$INSTANCE_POOL_DETAILS" ] || [ "$INSTANCE_POOL_DETAILS" == "null" ]; then
+  echo "No instance pool found with name $INSTANCE_POOL_NAME. Using default size $INSTANCE_POOL_SIZE..."
+else
+  export INSTANCE_POOL_SIZE=$(echo "$INSTANCE_POOL_DETAILS" | jq -r '.size')
+  echo "Found existing instance pool with name $INSTANCE_POOL_NAME.  Using existing size $INSTANCE_POOL_SIZE"
+fi
+
+# run as user
+if [ -z "$1" ]; then
+  SSH_USER=$(whoami)
+  echo "Ansible SSH user is not defined. We use current user: $SSH_USER"
+else
+  SSH_USER=$1
+  echo "Run ansible as $SSH_USER"
+fi
+
+[ -z "$USER_PUBLIC_KEY_PATH" ] && USER_PUBLIC_KEY_PATH="~/.ssh/id_ed25519.pub"
+
+[ -z "$USER_PRIVATE_KEY_PATH" ] && USER_PRIVATE_KEY_PATH="~/.ssh/id_ed25519"
+
+[ -z "$POSTINSTALL_STATUS_FILE" ] && POSTINSTALL_STATUS_FILE="$(realpath $LOCAL_PATH/../../..)/test-results/nomad_postinstall_status.txt"
+
+[ -z "$S3_PROFILE" ] && S3_PROFILE="oracle"
+[ -z "$S3_STATE_BUCKET" ] && S3_STATE_BUCKET="tf-state-$ENVIRONMENT"
+[ -z "$S3_ENDPOINT" ] && S3_ENDPOINT="https://$ORACLE_S3_NAMESPACE.compat.objectstorage.$ORACLE_REGION.oraclecloud.com"
+[ -z "$S3_STATE_KEY" ] && S3_STATE_KEY="$ENVIRONMENT/nomad-whisper/$POOL_TYPE/terraform.tfstate"
+
+[ -z "$BASE_IMAGE_TYPE" ] && BASE_IMAGE_TYPE="$NOMAD_BASE_IMAGE_TYPE"
+[ -z "$BASE_IMAGE_TYPE" ] && BASE_IMAGE_TYPE="JammyBase"
+
+arch_from_shape $SHAPE
+
+[ -z "$IMAGE_OCID" ] && IMAGE_OCID=$($LOCAL_PATH/../../scripts/oracle_custom_images.py --type $BASE_IMAGE_TYPE --architecture "$IMAGE_ARCH" --region="$ORACLE_REGION" --compartment_id="$COMPARTMENT_OCID" --tag_namespace="$TAG_NAMESPACE")
+if [ -z "$IMAGE_OCID" ]; then
+  echo "No IMAGE_OCID found.  Exiting..."
+  exit 210
+fi
+
+[ -z "$AVAILABILITY_DOMAINS" ] && AVAILABILITY_DOMAINS=$(oci iam availability-domain list --region=$ORACLE_REGION | jq .data[].name | jq --slurp .)
+if [ -z "$AVAILABILITY_DOMAINS" ]; then
+  echo "No AVAILABILITY_DOMAINS found.  Exiting..."
+  exit 206
+fi
+
+VCN_NAME_ROOT="$ORACLE_REGION-$ENVIRONMENT"
+VCN_NAME="$VCN_NAME_ROOT-vcn"
+
+TERRAFORM_MAJOR_VERSION=$(terraform -v | head -1  | awk '{print $2}' | cut -d'.' -f1)
+TF_GLOBALS_CHDIR=
+TF_GLOBALS_CHDIR_SG=
+if [[ "$TERRAFORM_MAJOR_VERSION" == "v1" ]]; then
+  TF_GLOBALS_CHDIR="-chdir=$LOCAL_PATH"
+  TF_GLOBALS_CHDIR_SG="-chdir=$LOCAL_PATH/security-group"
+  TF_CLI_ARGS=""
+  TF_POST_PARAMS=
+  TF_POST_PARAMS_SG=
+else
+  TF_POST_PARAMS="$LOCAL_PATH"
+  TF_POST_PARAMS_SG="$LOCAL_PATH/security-group"
+fi
+
+# first find or create the nomad security group
+[ -z "$S3_STATE_KEY_NOMAD_SG" ] && S3_STATE_KEY_NOMAD_SG="$ENVIRONMENT/nomad-whisper/$POOL_TYPE/terraform-nomad-sg.tfstate"
+LOCAL_NOMAD_SG_KEY="terraform-nomad-sg.tfstate"
```

Comment:
Unless you are creating a new sg we don't need this part.

Reply:
Instead we should use the nomad shared security group

`
data "oci_core_network_security_groups" "nomad_network_security_groups" {
  compartment_id = var.compartment_ocid
  filter {
    name = "display_name"
    values = ["${var.environment}-${var.oracle_region}-nomad-pool-shared-SecurityGroup"]
  }
}`

Reply:
Example utilization:

https://github.com/jitsi/infra-provisioning/blob/05ff7a8ebe2704cc05ba6e8ff48b280bc8c00691/terraform/jibri-instance-configuration/create-jibri-instance-configuration.tf#L93